### ISSUE TITLE
[Add2App] Android leaks the FlutterView from 1.22.0-12.0.pre

### ISSUE THREAD

**Issue by dannyvalentesonos** [flutter/flutter#65941]
A new leak has appeared in Flutter **1.22.0-12.0.pre** which wasn't present in **1.22.0-9.0.pre**

![mem_leak_1 22 0-12](https://user-images.githubusercontent.com/40898687/93347152-c2338200-f802-11ea-83de-26a71c8a8db8.png)

A sample app that reproduces this:
[Add2AppTest-1.22.0-12.0.pre-dev.zip](https://github.com/flutter/flutter/files/5232598/Add2AppTest-1.22.0-12.0.pre-dev.zip)

[✓] Flutter (Channel dev, 1.22.0-9.0.pre, on Mac OS X 10.15.6 19G2021, locale en-US)
    • Flutter version 1.22.0-9.0.pre at /Users/danny.valente/Library/flutter/1.22.0-9.0.pre-dev
    • Framework revision 7a43175198 (3 weeks ago), 2020-08-28 23:18:04 -0400
    • Engine revision 07e2520d5d
    • Dart version 2.10.0 (build 2.10.0-73.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny.valente/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

**Comment by dannyvalentesonos** [comment]
A new leak is encountered in the Android Add2App API very often. Almost every time we update the Flutter SDK. Is there any test coverage around this?

**Comment by TahaTesser** [comment]
<img src="https://user-images.githubusercontent.com/48603081/93443394-e8821c00-f8ed-11ea-9f37-dc075e7b0ef0.png" width="300" height="600" />



<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel dev, 1.22.0-12.0.pre, on Microsoft Windows [Version 10.0.19041.508], locale en-US)
    • Flutter version 1.22.0-12.0.pre at C:\Code\flutter_dev
    • Framework revision a27c242b0e (5 days ago), 2020-09-11 17:46:57 -0400
    • Engine revision fee6f9e533
    • Dart version 2.10.0 (build 2.10.0-110.0.dev)


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.49.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.1

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Web Server (web)                   • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 85.0.4183.102

• No issues found!
```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.22.0-10.0.pre.221, on Microsoft Windows [Version 10.0.19041.508], locale en-US)
    • Flutter version 1.22.0-10.0.pre.221 at C:\Code\flutter_master
    • Framework revision fa646339d6 (8 hours ago), 2020-09-16 17:47:04 -0700
    • Engine revision 933f811d94
    • Dart version 2.10.0 (build 2.10.0-135.0.dev)


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Code\sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\Code\sdk
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30503.244
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.0)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.49.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.1

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.508]
    • Web Server (web)                   • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 85.0.4183.102

• No issues found!
```

</details>

**Comment by xster** [comment]
Looks like https://github.com/flutter/engine/pull/20736 (which takes a view but doesn't release it). Might be a bit to revert. @GaryQian could you take a look? We should cherry-pick this.

**Comment by chinmaygarde** [comment]
cc @jason-simmons

**Comment by dannyvalentesonos** [comment]
We are trying to get onto a stable build, and we were targeting 1.22. Is there any chance to get this fix in 1.22?

**Comment by xster** [comment]
@dannyvalentesonos depending on the size of the patch. But we agree it's important to fix this in 1.22 ideally.

**Comment by xster** [comment]
Thanks for the fix @jason-simmons!
